source/common/router/router.h

```diff
@@ -355,6 +355,9 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     bool upstream_canary_ : 1;
     bool encode_complete_ : 1;
     bool encode_trailers_ : 1;
+    // Tracks whether we deferred a per try timeout because the downstream request
+    // had not been completed yet.
+    bool pending_per_try_timeout_ : 1;
```

Comment:
nit: I would probably name this something like `create_per_try_timeout_on_request_complete_` or something like that to differentiate between waiting for the per try timeout to  elapse which is what this sounds like to me.